Automattic/jetpack
# Connection: Improve JS code in the connection owner removal notice

## Proposed changes:
* Replace jQuery code with VanillaJS
* Address feedback from #28516

### Other information:


## Jetpack product discussion
1204007051815258-as-1203980469022872

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect Jetpack, create another user (non-connected).
2. Log in as the second user, log into WPCOM under different account.
3. Go to "Users", click "Delete" for the connection owner.
4. You should see the Jetpack notice. Compare it to the screenshot below. <img width="822" alt="Screen Shot 2023-01-26 at 19 34 40" src="https://user-images.githubusercontent.com/1341249/214981445-d5af5d0f-5bd7-412a-9d5c-a0f8f97be815.png">
5. Click "Connect to Jetpack". Finish connection, reload the "Delete user" page.
6. The notice show now suggest you to replace the connection owner with another user. Compare it to the screenshot below. <img width="822" alt="Screen Shot 2023-01-26 at 19 38 15" src="https://user-images.githubusercontent.com/1341249/214981454-57d234a8-6d2c-4486-9b40-803d635a29db.png">
7. Choose a new connection owner, click "Set new connection owner". The notice should disappear, you can now delete the user.
8. Delete the user, confirm that you see no other notices.
9. Confirm connection ownership got transferred (you can use Jetpack Backup to see the connection owner ID).
